autoload/lsp/omni.vim

```diff
@@ -303,50 +303,64 @@ function! lsp#omni#get_vim_completion_items(options) abort
+                let l:text_edit_new_text = l:text_edit['newText']
+                let l:text_edit_filter_text = get(l:completion_item, 'filterText', v:null)
+                if !empty(l:text_edit_filter_text) && matchstr(l:text_edit_new_text, '^' . l:refresh_pattern) ==# ''
+                    " Use filterText as word.
+                    let l:vim_complete_item['word'] = l:text_edit_filter_text
+                else
+                    " Use textEdit.newText as word.
+                    let l:vim_complete_item['word'] = l:text_edit_new_text
                 endif
+
+                " Fix overlapped text if needed.
+                let l:item_start_character = l:range['start']['character']
+                if l:item_start_character < l:default_start_character
+                    " Add already typed word. The typescript-language-server returns `[Symbol]` item for the line of `Hoo.|`. So we should add `.` (`.[Symbol]`) .
+                    let l:overlap_text = strcharpart(l:current_line, l:item_start_character, l:default_start_character - l:item_start_character)
+                    if stridx(l:vim_complete_item['word'], l:overlap_text) != 0
+                        let l:vim_complete_item['word'] = l:overlap_text . l:vim_complete_item['word']
+                    endif
+                endif
+                let l:start_character = min([l:item_start_character, l:start_character])
+                let l:start_characters += [l:item_start_character]
+            endif
+        endif
+
+        if !has_key(l:vim_complete_item, 'word')
+            " Use insertText
+            if !empty(get(l:completion_item, 'insertText'))
+                let l:vim_complete_item['word'] = l:completion_item['insertText']
+                let l:start_characters += [l:default_start_character]
+
+            " Use label
+            else
+                let l:vim_complete_item['word'] = l:completion_item['label']
+                let l:start_characters += [l:default_start_character]
             endif
-            let l:start_character = min([l:item_start_character, l:start_character])
-            let l:start_characters += [l:item_start_character]
-        elseif has_key(l:completion_item, 'insertText') && !empty(l:completion_item['insertText'])
-            let l:vim_complete_item['word'] = l:completion_item['insertText']
-            let l:start_characters += [l:default_start_character]
-        else
-            let l:vim_complete_item['word'] = l:completion_item['label']
-            let l:start_characters += [l:default_start_character]
         endif
 
-        if l:expandable
-            let l:vim_complete_item['word'] = lsp#utils#make_valid_word(substitute(l:vim_complete_item['word'], '\$[0-9]\+\|\${\%(\\.\|[^}]\)\+}', '', 'g'))
+        if get(l:completion_item, 'insertTextFormat', 1) == 2
+            " for snippet
+            let l:vim_complete_item['word'] = lsp#utils#make_valid_word(l:vim_complete_item['word'])
```

Comment:
Remove `substitute(var, "'\$[0-9]\+\|\${\%(\\.\|[^}]\)\+}'", 'g')` because it is duplicated to 
https://github.com/prabirshrestha/vim-lsp/blob/c0482f0fb407c29a88875616175784a886f131c3/autoload/lsp/utils.vim#L450